### DISCUSSION TITLE
More generalized way to move the cursor in VM mode

### DISCUSSION THREAD

**Discussion by IndianBoy42** [mg979/vim-visual-multi#230] [Ideas]
For now `<Up><Down><Left><Right>` move the cursor, which potentially allows moving the cursor to an arbitrary position and create an arbitrary new selection (visual select and then `vm-visual-add`). This is actually not super clear in the docs, it says `<M-right><M-left>`. Am I mistaken about this somehow and its 'working' for a different reason?

Either way, this is kinda useful, but I miss being able to do other motions obviously. Is this an avenue of expansion that is being considered? I can think of a couple of ways to do this:

1. some key that allows you to do 1 motion and then 1 textobject/motion. not completely arbitrary but still powerful (combined with easymotion or leap or hop especially)
2. some key that exits visual-multi but keeps the highlights around (slightly dimmed maybe) and remembers the selections/patterns/registers etc then any future operation that would go to visual-multi will go back into the existing visual-multi session. ie implicitly calls `reselect-last`

**Comment by mg979** [comment]
There is `Toggle Mappings` plug for this reason, check the docs.

**Reply by IndianBoy42** [reply]
Ah...of course it is, i swear i just was reading the docs and somehow didn't realize it, that's exactly what I wanted
